drivers/adc/adc_mcux_lpadc.c

```diff
@@ -195,7 +195,8 @@ static void mcux_lpadc_start_channel(const struct device *dev)
 	lpadc_conv_command_config_t cmd_config;
 
 	LPADC_GetDefaultConvCommandConfig(&cmd_config);
-	cmd_config.channelNumber = data->channel_id;
+	cmd_config.channelNumber = data->channel_id & 0b111;
```

Comment:
@JeroenReeskamp, I think you want to use the ADC_CMDL_ADCH() macro defined in the  NXP HAL instead of a hard-coded 0b111.

Reply:
Maybe I'm missing something, but this macro returns channel 8 when I pass a value of 8. This does not work as I expect it to return channel 0 when I pass a value of 8.

Reply:
The driver has A/B sides which each have 8 channels as far as I can see. Maybe make that clear with a define, for example:

```c
#define CHANNELS_PER_SIDE 8
// ....

cmd_config.channelNumber = data->channel_id % CHANNELS_PER_SIDE;
cmd_config.sampleChannelMode = (data->channel_id < CHANNELS_PER_SIDE) ?
		kLPADC_SampleChannelSingleEndSideA :
		kLPADC_SampleChannelSingleEndSideB;
```